tests/v2/validation/provisioning/k3s/secret_encryption_test.go

```diff
@@ -0,0 +1,320 @@
+package k3s
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/secrets"
+	"github.com/rancher/rancher/tests/framework/extensions/sshkeys"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+type SecretEncryptionTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+	psact                 string
+	advancedOptions       provisioning.AdvancedOptions
+}
+
+const (
+	etcdLabel         = "node-role.kubernetes.io/etcd"
+	machineAnnotation = "cluster.x-k8s.io/machine"
+	externalIP        = "k3s.io/external-ip"
+	secretKey         = "testKey"
+	secretValue       = "testValue"
+	etcdctlInstallCmd = `ETCD_VER=v3.5.0 && GOOGLE_URL=https://storage.googleapis.com/etcd && GITHUB_URL=https://github.com/etcd-io/etcd/releases/download &&
+	DOWNLOAD_URL=${GOOGLE_URL} && rm -f /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz && rm -rf /tmp/etcd-download-test && mkdir -p /tmp/etcd-download-test &&
```

Comment:
Some if not all of these consts should be in extensions.

Reply:
I agree. I think some of them definitely seem useful